core/src/main/scala/kafka/server/metadata/ClientQuotaMetadataManager.scala

```diff
@@ -115,36 +117,28 @@ class ClientQuotaMetadataManager(private[metadata] val quotaManagers: QuotaManag
       case _ => throw new IllegalStateException("Should only handle IP quota entities here")
     }
 
-    // The connection quota only understands the connection rate limit
-    if (quotaRecord.key() != QuotaConfigs.IP_CONNECTION_RATE_OVERRIDE_CONFIG) {
-      warn(s"Ignoring unexpected quota key ${quotaRecord.key()} for entity $ipEntity")
-      return
-    }
-
-    // Convert the value to an appropriate Option for the quota manager
-    val newValue = if (quotaRecord.remove()) {
-      None
-    } else {
-      Some(quotaRecord.value).map(_.toInt)
-    }
-    try {
-      connectionQuotas.updateIpConnectionRateQuota(inetAddress, newValue)
-    } catch {
-      case t: Throwable => error(s"Failed to update IP quota $ipEntity", t)
+    quotaDelta.changes().entrySet().forEach { e =>
+      // The connection quota only understands the connection rate limit
+      if (!e.getKey().equals(QuotaConfigs.IP_CONNECTION_RATE_OVERRIDE_CONFIG)) {
```

Comment:
nit: it would make the code a little more readable if use used vals with good names for `e.getKey` and `e.getValue`